aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/RunFaceLivenessSession.kt

```diff
@@ -29,41 +30,45 @@ import com.amplifyframework.predictions.aws.models.FaceTargetChallengeResponse
 import com.amplifyframework.predictions.aws.models.FaceTargetMatchingParameters
 import com.amplifyframework.predictions.aws.models.InitialFaceDetected
 import com.amplifyframework.predictions.aws.models.RgbColor
+import com.amplifyframework.predictions.aws.models.liveness.ChallengeConfig
 import com.amplifyframework.predictions.aws.models.liveness.FreshnessColor
+import com.amplifyframework.predictions.aws.models.liveness.OvalParameters
 import com.amplifyframework.predictions.aws.models.liveness.SessionInformation
+import com.amplifyframework.predictions.models.Challenge
 import com.amplifyframework.predictions.models.ChallengeResponseEvent
+import com.amplifyframework.predictions.models.ChallengeType
+import com.amplifyframework.predictions.models.FaceLivenessChallengeType
 import com.amplifyframework.predictions.models.FaceLivenessSession
 import com.amplifyframework.predictions.models.FaceLivenessSessionChallenge
 import com.amplifyframework.predictions.models.FaceLivenessSessionInformation
 import com.amplifyframework.predictions.models.VideoEvent
 
 internal class RunFaceLivenessSession(
-    sessionId: String,
-    sessionInformation: FaceLivenessSessionInformation,
+    private val sessionId: String,
+    private val clientSessionInformation: FaceLivenessSessionInformation,
     val credentialsProvider: CredentialsProvider,
     livenessVersion: String?,
     onSessionStarted: Consumer<FaceLivenessSession>,
     onComplete: Action,
     onError: Consumer<PredictionsException>
 ) {
 
-    private val livenessEndpoint = "wss://streaming-rekognition.${sessionInformation.region}.amazonaws.com:443"
-
     private val livenessWebSocket = LivenessWebSocket(
         credentialsProvider = credentialsProvider,
-        endpoint = "$livenessEndpoint/start-face-liveness-session-websocket?session-id=$sessionId" +
-            "&challenge-versions=${sessionInformation.challengeVersions}&video-width=" +
-            "${sessionInformation.videoWidth.toInt()}&video-height=${sessionInformation.videoHeight.toInt()}",
-        region = sessionInformation.region,
-        sessionInformation = sessionInformation,
+        endpoint = buildWebSocketEndpoint(),
+        region = clientSessionInformation.region,
+        clientSessionInformation = clientSessionInformation,
         livenessVersion = livenessVersion,
-        onSessionInformationReceived = { sessionInformation ->
-            val challenges = processSessionInformation(sessionInformation)
+        onSessionInformationReceived = { serverSessionInformation ->
+            val challenges = processSessionInformation(serverSessionInformation)
+            val challengeType = processChallengeType()
```

Comment:
Shouldn't we just provide `onError` if we don't understand the challenge type

Reply:
I agree with this. I modified the flow and have it so that we'll call `onError` in the web socket class.